presto-main/src/test/java/com/facebook/presto/operator/TestingBlockBuilders.java

```diff
@@ -0,0 +1,282 @@
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildIntegerBlock(int positionCount, boolean allowNulls)
+    {
+        return createIntsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextInt())
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildSmallintBlock(int positionCount, boolean allowNulls)
+    {
+        return createTypedLongsBlock(
+                SMALLINT,
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextLong() % Short.MIN_VALUE)
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildBooleanBlock(int positionCount, boolean allowNulls)
+    {
+        return createBooleansBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : RANDOM.nextBoolean())
+                        .collect(Collectors.toList()));
+    }
+
+    Block buildVarcharBlock(int positionCount, boolean allowNulls, int maxStringLength)
+    {
+        return createStringsBlock(
+                IntStream.range(0, positionCount)
+                        .mapToObj(i -> allowNulls && i % 7 == 0 ? null : getRandomString(maxStringLength))
+                        .collect(Collectors.toList()));
+    }
+
+    DictionaryBlock buildDictionaryBlock(Block dictionary, int positionCount)
+    {
+        int[] ids = IntStream.range(0, positionCount).map(i -> RANDOM.nextInt(dictionary.getPositionCount() / 10)).toArray();
+        return new DictionaryBlock(dictionary, ids);
+    }
+
+    RunLengthEncodedBlock buildRleBlock(Block block, int positionCount)
+    {
+        checkArgument(block.getPositionCount() > 0);
+        return new RunLengthEncodedBlock(block.getRegion(block.getPositionCount() / 2, 1), positionCount);
+    }
+
+    Block buildDictRleBlock(Block block, int positionCount, Optional<String> wrappings)
+    {
+        if (!wrappings.isPresent()) {
+            return null;
+        }
+
+        for (int i = wrappings.get().length() - 1; i >= 0; i--) {
+            if (wrappings.get().charAt(i) == 'D') {
+                block = buildDictionaryBlock(block, positionCount);
+            }
+            else if (wrappings.get().charAt(i) == 'R') {
+                block = buildRleBlock(block, positionCount);
+            }
+            else {
+                throw new IllegalArgumentException(format("wrappings %s is incorrect", wrappings));
+            }
+        }
+        return block;
+    }
+
+    Block buildBlockWithType(Type type, int positionCount, boolean allowNulls, boolean isView, Optional<String> wrappings)
+    {
+        Block block = null;
+
+        if (isView) {
+            positionCount *= 2;
+        }
+
+        if (type == BIGINT || type instanceof DecimalType && ((DecimalType) type).isShort()) {
```

Comment:
it might be clearer to handle decimal type separately:

```
        Block block;
        if (type == BIGINT) {
            block = buildBigintBlock(positionCount, allowNulls);
        }
        else if (type == INTEGER) {
            block = buildIntegerBlock(positionCount, allowNulls);
        }
        else if (type == SMALLINT) {
            block = buildSmallintBlock(positionCount, allowNulls);
        }
        else if (type instanceof DecimalType) {
            DecimalType decimalType = (DecimalType) type;
            if (decimalType.isShort()) {
                block = buildBigintBlock(positionCount, allowNulls);
            }
            else {
                block = buildLongDecimalBlock(positionCount, allowNulls);
            }
        }
```